### ISSUE TITLE
Using with Redux-toolkit v2?

### ISSUE THREAD

**Issue by F-happy** [redux-observable/redux-observable#788]

**Comment by evertbouw** [comment]
Once we've published a compatibility update you can use Redux-Observable with Redux-toolkit like this https://github.com/evertbouw/vite-template-redux-observable

**Comment by ratoaq2** [comment]
will there be a release supporting the new redux major version?

**Comment by evertbouw** [comment]
If you aren't using typescript it should work with the existing version. I made the code changes so I think it's ready, but afaik only @jayphelps has the permissions to publish. I could publish a candidate under my own scope if you'd be ok using that.

**Comment by jayphelps** [comment]
@evertbouw I've added you to npm. Otherwise I should have time to do the publish next week when I'm off work. Busy times at work right now.

**Comment by evertbouw** [comment]
Ok I'll publish a release candidate later tonight

**Comment by evertbouw** [comment]
Published RC2 and brought the template under the redux-observable org https://github.com/redux-observable/vite-template-redux-observable

**Comment by glarget** [comment]
Hi, I have a question about RTK V2, Redux V5 and React-redux V9. 

With the new migration guide here https://redux.js.org/usage/migrations/migrating-rtk-2, I followed the instructions and I use redux-observable v2.

While yarn install, I saw this warning: **"> redux-observable@2.0.0" has incorrect peer dependency "redux@>=4 <5".**

Could my application work with redux-observable 2.0 combined with all the latest redux / RTK versions ?

If no: 

- Could I install your new 3.0.0-rc2 version and go in production with it ?  

Thank you very much for your help !

**Comment by evertbouw** [comment]
@glarget yes you can install `3.0.0-rc2`. I've not received any bug reports so I should publish 3.0.0 soon without any changes.

**Comment by azangru** [comment]
It's been 3 weeks since the last comment. When will soon come? 🥺

**Comment by azangru** [comment]
It's been a month and a half now... 😞

**Comment by jayphelps** [comment]
@azangru entitlement in open source isn't a good look. Use some other library or make your own if this is bothering you.

**Comment by jayphelps** [comment]
@azangru What I said was a bit harsh, I'm sorry about that. Said another way: we (the maintainers) are volunteers and none of us actually work on any apps that use Redux Observable at the moment. I don't even maintain _any_ UI application right now as part of my job. So it's hard to find time, and for me specifically it's hard to find motivation to _make_ the time because I haven't really enjoyed doing open source for many years after being burnt out from people's thankless demands. In this case 3.0.0-rc2 is available, so it's frustrating to see your comments which imply something is blocking you, when I don't believe anything is?

**Comment by evertbouw** [comment]
tagging a new version number doesn't really change anything but it can go wrong, so I need to take the time for this and actually do it. Getting the website in order is a higher priority for me but would take even more time. Having the latest tagged as 'release candidate' hopefully directs people to the git repo for the latest information.

the release candidate is stable, please use it.

**Comment by mark-dr** [comment]
Thank you @jayphelps and @evertbouw for keeping redux-observable up to date :heart: It's a critical library for so many of us.

**Comment by simonepizzamiglio** [comment]
@evertbouw I’m working on a monorepo that uses Redux Toolkit v2 with redux-observable v2. While everything works fine, there’s a dependency mismatch: redux-observable v2 relies on Redux v4, but Redux Toolkit v2 depends on Redux v5. This is causing duplicated Redux versions in the node_modules folder.

To resolve this, I’m trying to update redux-observable to v3 which uses Redux v5. However, I’m encountering several TypeScript issues, particularly around how the `Epic` type is defined in v3.

Specifically, I’m curious about the reasoning behind [this change](https://github.com/redux-observable/redux-observable/pull/785/files#diff-f0de5c654074b6f685ac0fd941b3193cc9dd48700a9e07e29755d4d2c99308deL6-R5). In previous versions, the `Epic` type extended `Action`, but now it defaults to `unknown`.

Could you explain why it no longer extends `Action`? What was the rationale behind this change?

**Comment by evertbouw** [comment]
@simonepizzamiglio redux uses `unknown` as the action type now, so I thought it would be best to keep that uniform. https://redux.js.org/usage/migrations/migrating-rtk-2#middleware-type-changed---middleware-action-and-next-are-typed-as-unknown

**Comment by simonepizzamiglio** [comment]
``

> @simonepizzamiglio redux uses `unknown` as the action type now, so I thought it would be best to keep that uniform. https://redux.js.org/usage/migrations/migrating-rtk-2#middleware-type-changed---middleware-action-and-next-are-typed-as-unknown

@evertbouw makes sense! Thanks for answering! What about the `State` generic type to default to `void`?

**Comment by evertbouw** [comment]
@simonepizzamiglio it has been `void` for the middleware, and `any` for the Epic type, I think in my quest to get rid of anys I made both of them `void`. Perhaps `unknown` is better?

**Comment by simonepizzamiglio** [comment]
@evertbouw `unknown` seems safer than `void` in this case. A state might exist, but what if the user doesn’t explicitly define it in the `Epic` type? Using `void` assumes the state doesn’t exist, correct?

**Comment by evertbouw** [comment]
@simonepizzamiglio yeah it is meant to be explicitly provided to the type (that you can reexport so you don't have to repeat it all the time)

**Comment by MikeRyanDev** [comment]
This is a pretty dated issue. We want to get this package working against the latest Redux and Redux Toolkit. I'm going to close this for now unless there's something specific I'm missing.
